src/processing/strategies/transform.rs

```diff
@@ -0,0 +1,38 @@
+use crate::processing::strategies::{CommitRequest, MessageRejected, ProcessingStrategy};
+use crate::types::Message;
+
+pub struct Transform<TPayload: Clone + Send + Sync, TTransformed: Clone + Send + Sync> {
+    pub function: fn(Message<TPayload>) -> TTransformed,
+    pub next_step: Box<dyn ProcessingStrategy<TTransformed>>,
+}
+
+impl<TPayload: Clone + Send + Sync, TTransformed: Clone + Send + Sync> ProcessingStrategy<TPayload>
+    for Transform<TPayload, TTransformed>
+{
+    fn poll(&mut self) -> Option<CommitRequest> {
+        None
+    }
+
+    fn submit(&mut self, message: Message<TPayload>) -> Result<(), MessageRejected> {
+        let transformed = (self.function)(message.clone());
+
+        let partition = message.partition;
+        let offset = message.offset;
+        let timestamp = message.timestamp;
+
+        self.next_step.submit(Message {
```

Comment:
How is `MessageRejected` supposed to be handled? Does it keep getting passed up the chain and the first one handles it?

Reply:
None of the strategies need to handle it. The StreamProcessor should handle it and retry the message after some time.